Project-Spring-2-Flying-Fisherman

Flying-Fisherman opened the pull request:
이번 달 도 잘 부탁드립니다.

주제 정하기 힘드네요...

hannut91 committed:
Initial commit

hannut91 committed:
First Commit

Flying-Fisherman committed:
주제만 하루동안 고민 중

johngrib reviewed (commented):
.gradle 이나 .idea처럼 커밋할 필요 없는 파일들이 많이 들어간 것 같아요.

Flying-Fisherman committed:
주제 지정

Flying-Fisherman committed:
`.gitattributes`, `.gitignore`, `App.java` 생성

Flying-Fisherman committed:
Schedule Entity 및 ScheduleRepository 생성

Flying-Fisherman committed:
Project Structure 재구성

Flying-Fisherman committed:
ScheduleRepository, ScheduleService 생성 및 테스트 작성 시작

Flying-Fisherman commented:
작성 중간에 Structure이 이상하게 설정되어 있어서 재설정 하는데 꽤나 애먹었습니다...
덕분에 Service 쪽을 간략하게 구성하고 TDD 식으로 해보려는데 오류가 나버려서 그제서야 잘못된걸 깨달아버렸네요

johngrib commented:
> 작성 중간에 Structure이 이상하게 설정되어 있어서 재설정 하는데 꽤나 애먹었습니다...
> 덕분에 Service 쪽을 간략하게 구성하고 TDD 식으로 해보려는데 오류가 나버려서 그제서야 잘못된걸 깨달아버렸네요

아쉬운 일이지만 이런 작은 일들도 경험이 될 거라 생각해보면 어떨까요 :-)

johngrib commented on app/src/main/java/com/codesoom/scheduleMaker/application/ScheduleService.java:
```diff
@@ -0,0 +1,41 @@
+
+@Service
+@Transactional
+public class ScheduleService {
```
읽는 사람을 위한 JavaDoc을 잊지 말아주세요.

Flying-Fisherman committed:
`ScheduleData` DTO 작성

Flying-Fisherman committed:
`ScheduleNotFoundException` 생성

Flying-Fisherman committed:
`ScheduleService` CRUD 생성 및 Test 작성

- CRUD 구동에 필요한 일부 메서드를 `Schedule`, `ScheduleServide`에서 생성하였습니다.
- Test를 기반으로 하여 `ScheduleService`를 구성하였습니다.

Flying-Fisherman committed:
`ScheduleService` JavaDoc 작성

johngrib reviewed (commented):
이 repository는 코드 리뷰 용도가 아니라 프로젝트 용도이기 때문에 PR 하나에 계속해서 커밋을 추가하지 않으셔도 됩니다. 작업별로 PR을 올리시고, 직접 머지하시는 방식으로 진행해 보세요.

johngrib commented on app/src/test/java/com/codesoom/scheduleMaker/application/ScheduleServiceTest.java:
```diff
@@ -0,0 +1,126 @@
+        });
+    }
+
+    @Test
```
각 테스트에 설명이 없어서 어떤 테스트를 하고 있는지 한 눈에 알기가 어려워요.

johngrib commented on app/src/main/java/com/codesoom/scheduleMaker/application/ScheduleService.java:
```diff
@@ -0,0 +1,97 @@
+     *
+     * @param id Schedule 식별자
+     * @return 검색된 Schedule
+     * @throws ScheduleNotFoundException 검색 조건에 맞는 Schedule이 없을 시 예외를 던집니다.
```
```suggestion
     * @throws ScheduleNotFoundException 검색 조건에 맞는 Schedule이 없을 경우
```

johngrib commented on app/src/main/java/com/codesoom/scheduleMaker/application/ScheduleService.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    /**
+     * 식별자를 이용해 Schedule을 검색합니다.
```
```suggestion
     * 식별자를 이용해 Schedule을 검색해 리턴합니다.
```

Flying-Fisherman committed:
'ScheduleController' 생성 및 기타 JavaDoc 작성, 일부 오류 수정

- 'ScheduleController'을 생성하였습니다.
- 'Schedule', 'ScheduleData', 'ScheduleService' JavaDoc을 수정, 작성하였습니다.
- 일부 메서드에서 잘못 기입된 Data 를 수정하였습니다.
- `application.yml`을 추가하였습니다.

johngrib reviewed (changes requested):
`.idea`, `.gradle`은 관리 대상이 아니니 커밋하면 안됩니다. `.gitignore`에 추가하시고 `git rm`으로 삭제해주세요.

Flying-Fisherman committed:
`.gradle`, `.idea` 삭제

Flying-Fisherman committed:
`User` 서비스 및 테스트 작업, `Schedule` 컨트롤러 및 `ErrorAdvice` 수정

- `ScheduleControllerTest` 추가 및 수정
- `ControllerErrorAdvice` 및 기타 ErrorHandler 작성 및 추가
- `User` Entity 작성 및 관련 DTO 작성
- 'UserService' 테스트 및 구축 중
- JavaDoc 꾸준히 작성 중

johngrib reviewed (commented):
이 세가지 작업이 모두 하나의 커밋으로 만들어졌어요.
따로따로 커밋하는 연습이 필요해요.

![image](https://user-images.githubusercontent.com/1855714/124384448-43b1bd80-dd0c-11eb-9a84-eaeb81f7396e.png)

Flying-Fisherman committed:
`UserController`, `UserService` 구성 완료

- User Service, Controller, Repository Layer 구성
(추후 개선)
- 관련 Test 완료

Flying-Fisherman committed:
Login 및 암호화 동작 작성

- Login 관련 `SessionController`, `AuthenticationService` 밑 기타 로직 작성 완료

- 위 사항에 대한 에러 구현 완료

- Test는 아직 Service Layer 에서만 작성

ps. TDD로 해보려고 생각하지만 자꾸 BDD로 가버리는...

hannut91 commented:
작업이 완료되었다면, 머지 후 새로운 작업을 만들어주세요

Flying-Fisherman commented:
2가지 질문이 있습니다.

1.  Merge 부분은 제 Repo에서 진행하고 같은 Branch에서 계속 Commit 넣고 Merge하고 이어가는게 맞는지 궁금합니다.
Git도 그렇고 GitHub도 자연스럽게 굴려본 경험이 없어서 이 부분이 저에게 좀 애매합니다...

2. 이제 프론트도 구성해야하는데 이 전 강의에서 설명해주신 Marko를 꼭 이용해야 하는지 궁금합니다
(역시 경험 부재로 인해 사용법을 단시간에 익히기가 부담이 될거 같다고 생각됩니다)

hannut91 commented:
1. Merge 부분은 제 Repo에서 진행하고 같은 Branch에서 계속 Commit 넣고 Merge하고 이어가는게 맞는지 궁금합니다.
Git도 그렇고 GitHub도 자연스럽게 굴려본 경험이 없어서 이 부분이 저에게 좀 애매합니다...

=> 지금 레포에서 진행하고, 새로운 branch를 만들어서 작업을 완료한 후 `main` branch로 Merge하고 이어가는게 맞습니다. 그래서 하나의 풀리퀘스트가 하나의 작업이 되고, 이 작업을 위한 커밋들을 같이 올리고 머지합니다. 실무에서는 이렇게 작업이 진행됩니다. 처음에는 적응하는게 어려우실 수 있으나, 몇번 하다보면 익숙해져서 금방 배우실 수 있으실거에요. 

2. 이제 프론트도 구성해야하는데 이 전 강의에서 설명해주신 Marko를 꼭 이용해야 하는지 궁금합니다
(역시 경험 부재로 인해 사용법을 단시간에 익히기가 부담이 될거 같다고 생각됩니다)

=> 편하신 것으로 진행하시면 됩니다! 리액트로 하신다면 제가 도와드릴 수 있습니다

Flying-Fisherman commented:
그럼 이 PR도 머지되면 새로 다시 올려야겠네요?

hannut91 commented:
맞습니다 ㅎㅎ 머지하고 새로운 브렌치를 만들어서 작업하고 푸시해서 새로운 풀 리퀘스트를 만드시면 됩니다!

Flying-Fisherman commented:
알겠습니다!
